File: my_ai/counting_evaluator.py

class CountingEvaluator():
    def __init__(self, player):
        self.__player = player
        
    def evaluate(self, position):
        grid = position.get_grid()
        #print(player)
        result = 0
        for row in grid:
            for cell in row:
                if cell != None:
                    result += 1 if cell == self.__player else -1
        #if len(position.get_valid_moves()) == 0:
        #    if result > 0:
        #        result = float('100')
        #    else:
        #        result = -float('100')
        return result

File: java/src/CountingEvaluator.java
public class CountingEvaluator {
    private char player;

    public CountingEvaluator(char player) {
        System.err.println("Evaluator created, player: " + player);
        this.player = player;
    }

    public int evaluate(Position position) {
        char[][] grid = position.getGrid();
        int result = 0;
        for(char[] row: grid) {
            for(char cell: row) {
                result += cell == 'E' ? 0 : (cell == this.player ? 1 : -1);
            }
        }
        return result;
    }
}